frontends/ast/genrtlil.cc

```diff
@@ -1840,8 +1840,8 @@ RTLIL::SigSpec AstNode::genRTLIL(int width_hint, bool sign_hint)
 			int width = max(width_hint, 1);
 			width_hint = -1, sign_hint = true;
 			children[0]->detectSignWidthWorker(width_hint, sign_hint);
-			children[1]->detectSignWidthWorker(width_hint, sign_hint);
 			RTLIL::SigSpec left = children[0]->genRTLIL(width_hint, sign_hint);
+			children[1]->detectSignWidthWorker(width_hint, sign_hint);
```

Comment:
I'm a bit surprised at by this change. It seems like we probably need to account for the sign and width of both sides of the comparison before generating the RTLIL for _either_ side. The existing code should produce the Verilog semantics we're looking for: the comparison is done on signed operands with _both_ are signed, and using the maximum of the widths of each side. I'm sure I'm missing something here. Can you help me understand?